llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -29274,6 +29274,11 @@ SDValue AArch64TargetLowering::LowerFixedLengthVECTOR_SHUFFLEToSVE(
     }
   }
 
+  // Try to widen the shuffle before generating a possibly expensive SVE TBL.
```

Comment:
Could there be any value for the patterns above to move this up a bit so that it's executed earlier in the function?

Reply:
I don't think it changes the result (as all the matching above is looking for single-instruction replacements). Also, I placed it here to be consistent with the non-SVE lowering, which attempts this just before generating the Neon TBL too.